src/Env.ml

```diff
@@ -20,6 +20,10 @@ module Make (Key : sig
   include Equal.S with type t := t
 
   val context_kind : string
+
+  val to_string : t -> string
+end) (Error : sig
+  val makeError : Key.t * string -> [> error ]
```

Comment:
Surprisingly for me, arguments of module functor in Ocaml can depend on each other.
So I can throw from `Env` errors with specific `Id` that allowed us to get rid of making `Id` from the string when we catch an error somewhere above and try to look up.